dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPConfigConstants.java

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.netty.internal.tcp;
+
+/**
+ * This interface describes the constants used for configuring a TCP Channel.
+ * 
+ * Adapted from {@link com.ibm.wsspi.tcpchannel.TCPConfigConstants}
+ * 
+ */
+public interface TCPConfigConstants {
+
+    /**
+     * The external name used for this TCP channel
+     */
+    String EXTERNAL_NAME = "externalName";
+
+    /**
+     * Host name that an inbound channel will use for listening
+     */
+    String HOST_NAME = "hostname";
+
+    /**
+     * port on which the inbound channel will listen
+     */
+    String PORT = "port";
+
+    /**
+     * The maximum number of concurrent connections allowed for an inbound channel
+     */
+    String MAX_CONNS = "maxOpenConnections";
+
+    /**
+     * An IPv4/IPv6 address exclusion list that an inbound channel will use when
+     * accepting new connections
+     */
+    String ADDR_EXC_LIST = "addressExcludeList";
+
+    /**
+     * A host name exclusion list that an inbound channel will use when accepting
+     * new connections
+     */
+    String NAME_EXC_LIST = "hostNameExcludeList";
+
+    /**
+     * An IPv4/IPv6 address inclusion list that an inbound channel will use when
+     * accepting new connections
+     */
+    String ADDR_INC_LIST = "addressIncludeList";
+
+    /**
+     * A host name inclusion list that an inbound channel will use when accepting
+     * new connections
+     */
+    String NAME_INC_LIST = "hostNameIncludeList";
+
+    /**
+     * The default time out for TCP operations for this channel
+     */
+    String INACTIVITY_TIMEOUT = "inactivityTimeout";
+
+    /**
+     * The size of the tcp socket sending buffer
+     */
+    String SEND_BUFF_SIZE = "sendBufferSize";
+
+    /**
+     * The size of the tcp socket receiving buffer
+     */
+    String RCV_BUFF_SIZE = "receiveBufferSize";
+
+    /**
+     * Minimum Port Value
+     */
+    int PORT_MIN = 0;
```

Comment:
Can we ever make use of port 0 to get the OS to give us a free port in OpenLiberty and track it etc?

Reply:
both liberty and netty will get this behavior from the JDK https://docs.oracle.com/javase/7/docs/api/java/net/ServerSocket.html#ServerSocket%28int%29